mac 15-intel pypy flake on test_execution_summary

Been hammering CI today, and saw a new flake. Never seen this one fail before, but it does appear to be just another slow startup issue (sleep is only 0.1). So lets leave it for now, if it comes up again we can add rerun backoff behavior


<details><summary>Test execution Flake</summary>
```
=================================== FAILURES ===================================
____________________________ test_execution_summary ____________________________

temp_output_dir = '/private/var/folders/2z/bks_gyyd7g1g0kj25ltpgjwm0000gn/T/pytest-of-runner/pytest-0/test_execution_summary0/'

    def test_execution_summary(temp_output_dir: str) -> None:
        assert (
            run_duct_command(
                ["sleep", "0.1"],
                sample_interval=0.05,  # small enough to ensure we collect at least 1 sample
                report_interval=0.1,
                output_prefix=temp_output_dir,
            )
            == 0
        )
        with open(os.path.join(temp_output_dir, SUFFIXES["info"])) as info:
            info_dict = json.loads(info.read())
        execution_summary = info_dict["execution_summary"]
        # Since resources used should be small lets make sure values are roughly sane
>       assert execution_summary["average_pmem"] < 10
E       TypeError: '<' not supported between instances of 'NoneType' and 'int'

test/duct_main/test_execution.py:92: TypeError
------------------------------ Captured log call -------------------------------
INFO     con-duct:duct_main.py:112 duct 0.19.0.post42+ga689887 is executing 'sleep 0.1'...
INFO     con-duct:duct_main.py:113 Log files will be written to /private/var/folders/2z/bks_gyyd7g1g0kj25ltpgjwm0000gn/T/pytest-of-runner/pytest-0/test_execution_summary0/
INFO     con-duct:duct_main.py:186 Summary:
Exit Code: 0
Command: sleep 0.1
Log files location: /private/var/folders/2z/bks_gyyd7g1g0kj25ltpgjwm0000gn/T/pytest-of-runner/pytest-0/test_execution_summary0/
Wall Clock Time: 0.113 sec
Memory Peak Usage (RSS): -
Memory Average Usage (RSS): -
Virtual Memory Peak Usage (VSZ): -
Virtual Memory Average Usage (VSZ): -
Memory Peak Percentage: -%
Memory Average Percentage: -%
CPU Peak Usage: -%
Average CPU Usage: -%
=========================== short test summary info ============================
FAILED test/duct_main/test_execution.py::test_execution_summary - TypeError: '<' not supported between instances of 'NoneType' and 'int'
======== 1 failed, 343 passed, 2 skipped, 12 rerun in 151.81s (0:02:31) ========
py: exit 1 (153.45 seconds) /Users/runner/work/duct/duct> pytest -vv --cov-report=xml test pid=59968
  py: FAIL code 1 (260.67=setup[107.22]+cmd[153.45] seconds)
  evaluation failed :( (260.77 seconds)
Error: Process completed with exit code 1.
```
</details>

--- a/test/duct_main/test_execution.py
+++ b/test/duct_main/test_execution.py
@@ -75,12 +75,18 @@ def test_sanity_green(temp_output_dir: str) -> None:
     assert_expected_files(temp_output_dir)
 
 
-def test_execution_summary(temp_output_dir: str) -> None:
+@pytest.mark.flaky(reruns=5)
+def test_execution_summary(
+    request: pytest.FixtureRequest, temp_output_dir: str
+) -> None:
+    # Scale sleep time on retries to handle slow CI runners (PyPy, Mac)
+    attempt = getattr(request.node, "execution_count", 1)
+    sleep_time = 0.1 * attempt
     assert (
         run_duct_command(
-            ["sleep", "0.1"],
+            ["sleep", str(sleep_time)],
             sample_interval=0.05,  # small enough to ensure we collect at least 1 sample
-            report_interval=0.1,
+            report_interval=sleep_time,
             output_prefix=temp_output_dir,
         )
         == 0

